src/game/boe.actions.cpp

```diff
@@ -3743,11 +3757,11 @@ bool is_sign(ter_num_t ter) {
 	return false;
 }
 
-bool check_for_interrupt(){
+bool check_for_interrupt(std::string confirm_dialog){
 	using kb = sf::Keyboard;
 	bool interrupt = false;
 	sf::Event evt;
-	if(replaying && has_next_action() && next_action_type() == "handle_interrupt"){
+	if(replaying && confirm_dialog == "confirm-interrupt-special" && has_next_action() && next_action_type() == "handle_interrupt"){
```

Comment:
It looks like you made up a new dialog name that doesn't exist here – the old code was using "confirm-interrupt" but you changed it to "confirm-interrupt-special". Same thing in the header file for the default parameter value.